dali/pipeline/operator/builtin/external_source.h

```diff
@@ -157,51 +153,31 @@ class ExternalSource : public Operator<Backend> {
       std::lock_guard<std::mutex> busy_lock(busy_m_);
       tl_data_.AddBack(data);
       copy_to_storage_events_.AddBack(copy_to_storage_event);
-      data_in_tl_.push_back(true);
     }
     cv_.notify_one();
   }
 
-
   /**
    * @brief Sets the data that should be passed out of the op
    * on the next iteration.
    */
   template<typename SrcBackend>
-  inline void SetDataSource(const vector<Tensor<SrcBackend>> &t, cudaStream_t stream = 0) {
-    if (std::is_same<SrcBackend, GPUBackend>::value && std::is_same<Backend, CPUBackend>::value) {
-      std::string msg = "Incorrect Backends warning. Loading GPU-originated data into CPU "
-                        "ExternalSource operator is discouraged and might be inefficient.";
-      DALI_WARN(msg);
-    }
-    DALI_ENFORCE(OperatorBase::batch_size_ == static_cast<int>(t.size()),
-                 "Data list provided to ExternalSource needs to have batch_size length.");
-    // Note: If we create a GPU source, we will need to figure
-    // out what stream we want to do this copy in. CPU we can
-    // pass anything as it is ignored.
-    std::list<uptr_vt_type> data;
-    std::list<uptr_cuda_event_type> copy_to_storage_event;
-    {
-      std::lock_guard<std::mutex> busy_lock(busy_m_);
-      data = t_data_.GetEmpty();
-      copy_to_storage_event = copy_to_storage_events_.GetEmpty();
-    }
-
-    data.front()->resize(t.size());
-    for (size_t i = 0; i < t.size(); ++i) {
-      (*(data.front()))[i].Copy(t[i], stream);
-    }
-    if (std::is_same<SrcBackend, GPUBackend>::value) {
-      cudaEventRecord(*copy_to_storage_event.front(), stream);
-    }
+  inline void SetDataSource(const TensorList<SrcBackend> &tl, cudaStream_t stream = 0) {
```

Comment:
There's no overload that accepts TensorVector and that would be nice for completeness. Maybe this function is not necessary and `SetDataSourceHelper` should be renamed `SetDataSource`? Overload resolution will pick the right overload (generic or one accepting vector of tensors).

Reply:
Done